packages/react-storage/src/components/StorageBrowser/views/LocationsView/getLocationsViewTableData/getLocationsViewTableData.ts

```diff
@@ -43,6 +55,25 @@ export const getLocationsViewTableData = ({
               content: { text: getPermissionName(permission) },
             };
           }
+          case 'action': {
+            return location.type === 'OBJECT'
+              ? {
+                  key,
+                  type: 'button',
+                  content: {
+                    icon: 'download',
+                    ariaLabel: downloadLabel(location.prefix),
+                    onClick: () => {
+                      onDownload(location);
+                    },
+                  },
+                }
+              : {
+                  key,
+                  type: 'text',
+                  content: { text: '' },
+                };
```

Comment:
Just a thought for the future, we could provide this data through a callback on the interface that accepts `location` as a parameter and isolate the conditional logic